src/Analyzers/CSharp/Analyzers/UseUTF8StringLiteral/UseUTF8StringLiteralDiagnosticAnalyzer.cs

```diff
@@ -114,12 +119,17 @@ private void AnalyzeOperation(OperationAnalysisContext context, INamedTypeSymbol
             context.ReportDiagnostic(
                 DiagnosticHelper.Create(Descriptor, location, option.Notification.Severity, additionalLocations, properties: null));
         }
-
         internal static bool TryConvertToUTF8String(StringBuilder? builder, SegmentedList<byte> values)
         {
             for (var i = 0; i < values.Count;)
             {
-                var ros = GetBytesForNextRune(values, i);
+                // We only need max 4 elements for a single Rune
+                var count = Math.Min(values.Count - i, 4);
+
+                // Need to copy to a regular array to get a ROS for Rune to process
+                values.CopyTo(i, s_charArray, 0, count);
```

Comment:
The analyzer run concurrently. Isn't there a possibility that two separate array creations are analyzed at the same time, and both ends up racing here?

Reply:
Seems like if you just need 4 bytes here you could stackalloc them into a Span.